### DISCUSSION TITLE
Less secure Apps and SMTP Mailing (Gmail)

### DISCUSSION THREAD

**Discussion by flashedk** [pocketbase/pocketbase#6126] [Q&A] [answered]
Hi!
Since Google began to sunset SMTP Mailing via Username/Password and requiring OAuth to mail, i'm wondering how to do this now in Pocketbase? We are using Google Workspace.

Excited to hear what you are using.

**Comment by ganigeorgiev** [comment, marked as answer]
Isn't [App passwords](https://support.google.com/accounts/answer/185833?hl=en) still supported for Google Workspace accounts (_note that it is different from the old "Less secure apps" option_)?

It is not for Google Workspace and it is a little outdated but your can also check the Gmail instructions in https://github.com/pocketbase/pocketbase/discussions/458#discussioncomment-3651995.

If it doesn't work then you have essentially 3 options:

- Customize PocketBase by registering your own mail client via the [`OnMailerSend` hook](https://pocketbase.io/docs/go-event-hooks/#onmailersend) (_you can also use the Go Google Cloud SDK if you want to use the Gmail APIs_)
- Use a relay/middleware/proxy SMTP service
- Use a different mail send provider

Support for SMTP OAuth I guess could be added in PocketBase (for example by accepting a refresh token and then issue an access token before each mail send or something similar) but I don't have time at the moment to work on it.

If someone have the time and wants to work on this feel free to do it and I'll try to assist you if you have any questions regarding the PocketBase internals (the major components are the [`mailer` subpackage](https://github.com/pocketbase/pocketbase/tree/master/tools/mailer) and the [`BaseApp.NewMailClient` method](https://github.com/pocketbase/pocketbase/blob/9b4200a65cff7ac7fc41f257b2eb66dfabe0faba/core/base.go#L551)).

**Reply by flashedk** [reply]
You're right, App Passwords are still a think. Got it working with those. 

Will have a look next year if the SMTP OAuth is something i can provide a PR for.
